backend/logger/singleton.py

```diff
@@ -0,0 +1,7 @@
+class Singleton(type):
+    _instances = {}
+
+    def __call__(cls, *args, **kwargs):
+        if cls not in cls._instances:
+            cls._instances[cls] = super(Singleton, cls).__call__(*args, **kwargs)
+        return cls._instances[cls]
```

Comment:
There are __pycache__ files in this MR, which we should remove

Reply:
Yeah, need to update the .gitignore file to fix this